src/coreclr/tools/SuperFileCheck/Program.cs

```diff
@@ -0,0 +1,623 @@
+                    }
+
+                    var stdOut = new StringBuilder();
+                    var stdErr = new StringBuilder();
+
+                    proc.OutputDataReceived += (_, e) =>
+                    {
+                        if (e.Data == null)
+                        {
+                            outputWaitHandle.Set();
+                        }
+                        else
+                        {
+                            stdOut.AppendLine(e.Data);
+                        }
+                    };
+
+                    proc.ErrorDataReceived += (_, e) =>
+                    {
+                        if (e.Data == null)
+                        {
+                            errorWaitHandle.Set();
+                        }
+                        else
+                        {
+                            stdErr.AppendLine(e.Data);
+                        }
+                    };
+
+                    proc.BeginOutputReadLine();
+                    proc.BeginErrorReadLine();
+
+                    await proc.WaitForExitAsync();
+                    outputWaitHandle.WaitOne();
+                    errorWaitHandle.WaitOne();
+
+                    var exitCode = proc.ExitCode;
+                    return new FileCheckResult(exitCode, stdOut.ToString(), stdErr.ToString());
+                }
+            }
+            catch (Exception ex)
+            {
+                return new FileCheckResult(1, String.Empty, ex.Message);
+            }
+        }
+
+        /// <summary>
+        /// Get the method name from the method declaration.
+        /// </summary>
+        static string GetMethodName(MethodDeclarationSyntax methodDecl)
+        {
+            return
+                methodDecl.ChildTokens()
+                .OfType<SyntaxToken>()
+                .Where(x => x.IsKind(SyntaxKind.IdentifierToken)).First().ValueText;
+        }
+
+        static IEnumerable<SyntaxTrivia> GetDescendantSingleLineCommentTrivia(SyntaxNode node)
+        {
+            return
+                node
+                .DescendantTrivia()
+                .Where(x => x.IsKind(SyntaxKind.SingleLineCommentTrivia));
+        }
+
+        /// <summary>
+        /// Gather all syntactical method declarations whose body contains
+        /// FileCheck syntax.
+        /// </summary>
+        static MethodDeclarationInfo[] FindMethodsByFile(string filePath, string[] checkPrefixes)
+        {
+            var syntaxTree = CSharpSyntaxTree.ParseText(SourceText.From(File.ReadAllText(filePath)));
+            var root = syntaxTree.GetRoot();
+
+            var trivia =
+                GetDescendantSingleLineCommentTrivia(root)
+                .Where(x =>
+                {
+                    if (x.Token.Parent != null)
+                    {
+                        return !x.Token.Parent.Ancestors().Any(p => p.IsKind(SyntaxKind.MethodDeclaration) && p.Span.Contains(x.Span));
```

Comment:
Can you explain this, in particular the `Contains`?  How does a parent -not- contain x?  Are leading/trailing trivia accessible from the node but not in its Span?

Reply:
This is needed as a comment before the method declaration is considered a child of the method declaration syntax.
```csharp
// trivia1
public void M()
{
   // trivia2
}
```
Both `// trivia1` and `// trivia2` are descendants of `MethodDeclarationSyntax`.

We could use `SyntaxKind.Block` but I wanted it to be clear we are looking for trivia inside the method declaration.